simplefs/simplefs.go

```diff
@@ -1356,26 +1451,40 @@ func (k *SimpleFS) SimpleFSFolderEditHistory(
 	ctx context.Context, path keybase1.Path) (
 	res keybase1.FSFolderEditHistory, err error) {
 	ctx = k.makeContext(ctx)
-	t, tlfName, _, _, err := remoteTlfAndPath(path)
-	if err != nil {
-		return keybase1.FSFolderEditHistory{}, err
-	}
-	tlfHandle, err := libkbfs.GetHandleFromFolderNameAndType(
-		ctx, k.config.KBPKI(), k.config.MDOps(), tlfName, t)
-	if err != nil {
-		return keybase1.FSFolderEditHistory{}, err
-	}
-
-	// Get the root node first to initialize the TLF.
-	node, _, err := k.config.KBFSOps().GetRootNode(
-		ctx, tlfHandle, libkbfs.MasterBranch)
+	fb, _, err := k.getFolderBranchFromPath(ctx, path)
 	if err != nil {
 		return keybase1.FSFolderEditHistory{}, err
 	}
-	if node == nil {
+	if fb == (libkbfs.FolderBranch{}) {
 		return keybase1.FSFolderEditHistory{}, nil
 	}
 
 	// Now get the edit history.
-	return k.config.KBFSOps().GetEditHistory(ctx, node.GetFolderBranch())
+	return k.config.KBFSOps().GetEditHistory(ctx, fb)
+}
+
+var _ libkbfs.Observer = (*SimpleFS)(nil)
+
+// LocalChange implements the libkbfs.Observer interface for SimpleFS.
+func (k *SimpleFS) LocalChange(
+	_ context.Context, _ libkbfs.Node, _ libkbfs.WriteRange) {
+	// No-op.
+}
+
+// BatchChanges implements the libkbfs.Observer interface for SimpleFS.
+func (k *SimpleFS) BatchChanges(
+	ctx context.Context, changes []libkbfs.NodeChange, _ []libkbfs.NodeID) {
+	k.subscribeLock.RLock()
+	defer k.subscribeLock.RUnlock()
+	for _, nc := range changes {
+		fb := nc.Node.GetFolderBranch()
+		if fb == k.subscribeCurrFB {
```

Comment:
So if we get unmerged then we don't notify updates?

Reply:
You should still get updates for unmerged updates on this device, the same way FUSE does.  What made you think otherwise?

Reply:
It looks like you're only notifying that the path is updated if the folder-branch hasn't changed.

Reply:
Oh, when the TLF goes onto an unmerged branch, it (confusingly) doesn't change the actual `FolderBranch` it's referenced with.  That concept of a branch is unused today, but it was meant to indicate other historical views, or forks, of a given TLF.  So it might be useful for the time travel stuff, but for now all TLFs have `libkbfs.MasterBranch` in their `FolderBranch`, even when unmerged.

Reply:
Understood, thanks for clarifying.